src/lib.rs

```diff
@@ -222,6 +224,68 @@ impl<T: InputLength> InputLength for LocatedSpan<T> {
     }
 }
 
+impl<T: AtEof> AtEof for LocatedSpan<T> {
+    fn at_eof(&self) -> bool {
+        self.fragment.at_eof()
+    }
+}
+
+impl<T> InputTake for LocatedSpan<T> where Self: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>> {
+    fn take(&self, count: usize) -> Self {
+        self.slice(..count)
+    }
+
+    fn take_split(&self, count: usize) -> (Self, Self) {
+        (self.slice(count..), self.slice(..count))
+    }
+}
+
+impl<T> InputTakeAtPosition for LocatedSpan<T>
+where
+    T: InputTakeAtPosition + InputLength,
+    Self: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>> + Clone,
+{
+    type Item = <T as InputTakeAtPosition>::Item;
+
+    fn split_at_position<P>(&self, predicate: P) -> IResult<Self, Self, u32>
+    where
+        P: Fn(Self::Item) -> bool
+    {
+        self.fragment.split_at_position(predicate)
+            .map(|(_i, o)| {
+                let split = o.input_len();
+                (self.slice(split..), self.slice(..split))
+            })
+            .map_err(|e| {
+                match e {
+                    Err::Error(Context::Code(_, e)) |
+                    Err::Failure(Context::Code(_, e)) => Err::Error(Context::Code(self.clone(), e)),
+                    Err::Incomplete(needed) => Err::Incomplete(needed),
+                    _ => unimplemented!("nom_locate didn't think Err::List was used in InputTakeAtPosition"),
+                }
+            })
+    }
+
+    fn split_at_position1<P>(&self, predicate: P, e: ErrorKind<u32>) -> IResult<Self, Self, u32>
+    where
+        P: Fn(Self::Item) -> bool
+    {
+        self.fragment.split_at_position1(predicate, e)
+            .map(|(_i, o)| {
+                let split = o.input_len();
+                (self.slice(split..), self.slice(..split))
+            })
+            .map_err(|e| {
+                match e {
+                    Err::Error(Context::Code(_, e)) |
+                    Err::Failure(Context::Code(_, e)) => Err::Error(Context::Code(self.clone(), e)),
+                    Err::Incomplete(needed) => Err::Incomplete(needed),
+                    _ => unimplemented!("nom_locate didn't think Err::List was used in InputTakeAtPosition"),
+                }
+            })
```

Comment:
There are lots of code in common between `split_at_position1` and `split_at_position`. Do you think there is a room for a factorization thanks to a a macro or a private function?